### PR TITLE
Remove lang dependencies from langlibs

### LINKED ISSUE #42275: [Task]: Remove `io.ballerina.lang` dependencies from langlibs
### Description

`module-info.java` classes of langlib modules requires `io.ballerina.lang` module to get some constant values. Adding those constants directly to langlibs will eliminate those dependencies. 

### Describe your task(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #42275

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/utils/Lists.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * List util class for the langlib xml module.
+ *
+ * @since 2201.9.0
+ */
+public class Lists {
```
Comment (HindujaB) [inline langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/utils/Lists.java:28]: Shall we rename the class name and the method name to be more meaningful? 
something like `XmlUtils.getList()`?

File: langlib/lang.__internal/src/main/java/module-info.java [hunk 2]
```diff
@@ -1,5 +1,4 @@
 module io.ballerina.lang.internal {
     exports org.ballerinalang.langlib.internal;
-    requires io.ballerina.lang;
     requires io.ballerina.runtime;
 }
```
Comment (HindujaB) [inline langlib/lang.__internal/src/main/java/module-info.java:4]:
``` suggestion
}

```
Let's add new lines in other `module-info.java` files as well

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**